src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -192,27 +300,76 @@ static ucs_status_t uct_rdmacm_cm_create_dummy_qp(struct rdma_cm_id *id,
     return UCS_OK;
 }
 
+static ucs_status_t uct_rdamcm_cm_ep_create_qpn(uct_rdmacm_cm_ep_t *cep,
+                                                struct uct_rdmacm_cm_qp_context *ctx)
+{
+    ucs_status_t status;
+
+    if (ctx->reserved_qpn_support) {
+        status = uct_rdamcm_cm_ep_create_reserved_qpn(cep, ctx);      
+    } else {
+        /* create a dummy qp in order to get a unique qp_num to provide to librdmacm */
+        status = uct_rdmacm_cm_create_dummy_qp(cep->id, ctx->dummy_qp_ctx.cq,
+                                               &cep->qp);
+    }
+
+    if (status == UCS_OK) {
+        cep->flags |= UCT_RDMACM_CM_EP_QPN_CREATED;
+    }
+
+    return status;
+}
+
+static void uct_rdamcm_cm_ep_destroy_qpn(uct_rdmacm_cm_ep_t *cep)
+{
+    struct uct_rdmacm_cm_qp_context *ctx;
+    ucs_status_t status;
+
+    if (!(cep->flags & UCT_RDMACM_CM_EP_QPN_CREATED)) {
+        return;
+    }
+
+    status = uct_rdmacm_cm_get_qp_context(uct_rdmacm_cm_ep_get_cm(cep),
```

Comment:
why need to call uct_rdmacm_cm_get_qp_context both here and also inside uct_rdamcm_cm_ep_destroy_reserved_qpn()?
just pass the context to uct_rdamcm_cm_ep_destroy_reserved_qpn as parameter